### PR TITLE
lestarch: fixing missing autocodes with time role port

### PR DESCRIPTION
**Pull request by LeStarch**
| | |
|:---|:---|
|**_Originating Project/Creator_**| infrastructure |
|**_Affected Component_**| Autocoders |
|**_Affected Architectures(s)_**|  |
|**_Related Issue(s)_**|  |
|**_Has Unit Tests (y/n)_**| n |
|**_Builds Without Errors (y/n)_**| y |
|**_Unit Tests Pass (y/n)_**| y  |
|**_Documentation Included (y/n)_**| y  |

---
## Change Description

Adds in autocoding when a time role port is specified by-hand (not only calculated by existence of telemetry or events).

## Rationale

`this->timeGet();` should be available for use when without requiring events or channels such that users may easily request time for other purposes.

### CODE REVIEW

File: Autocoders/Python/src/fprime_ac/generators/visitors/ComponentVisitorBase.py [hunk 1]
```diff
@@ -442,6 +442,9 @@ def initPortFlags(self, obj, c):
         c.has_output_ports = len(c.output_ports) > 0
         c.has_typed_output_ports = len(c.typed_output_ports) > 0
         c.has_serial_output_ports = len(c.serial_output_ports) > 0
+        roles = [role for name, ptype, sync, priority, role, max_number in c.output_ports]
```
Comment (Joshua-Anderson) [inline Autocoders/Python/src/fprime_ac/generators/visitors/ComponentVisitorBase.py:445]: Optional: use `_` for unused for loop variables.

**Review by Joshua-Anderson — COMMENTED** [review]
Changes look good, but if it's possible adding an autocoder test case in `Autocoders/Python/test` covering this would be awesome.

File: Autocoders/Python/src/fprime_ac/generators/visitors/ComponentVisitorBase.py (hunk 1 above)
Comment (LeStarch) [reply Autocoders/Python/src/fprime_ac/generators/visitors/ComponentVisitorBase.py:445]: Using `_` here would be inconsistent with the rest of the file.  So I intentionally avoided that.

Comment (Joshua-Anderson) [reply Autocoders/Python/src/fprime_ac/generators/visitors/ComponentVisitorBase.py:445]: :cry: I'm surprised LGTM isn't complaining about it.

**Comment by LeStarch** [comment]
Unit test added.

**Review by Joshua-Anderson — APPROVED** [review]

**Pull request merged**